src/iris/bin/sender.py

```diff
@@ -558,6 +553,14 @@ def aggregate(now):
                 logger.info('[-] purged %s from messages %s remaining', active_message_ids, len(messages))
             del queues[key]
             sent[key] = now
+    inactive_message_ids = set(messages.keys()) - all_actives
```

Comment:
I think either of these would work instead of casting the .keys() list to a set: `set(messages)` or `messages.viewkeys()`